src/main/java/newuser/NewUserInteractor.java

```diff
@@ -1,4 +1,32 @@
 package newuser;
 
-public class NewUserInteractor {
+import database.UserDbGateway;
+import entities.FacilityUser;
+import entities.User;
+import entities.FacilityType;
+
+import java.util.UUID;
+
+
+public class NewUserInteractor{
+
+    UserDbGateway readWriter = new UserDbGateway();
+
+    public NewUserInteractor(){
+    }
+
+    public User addNewUser(String username, String password, UUID facilityID, FacilityType type){
+
+        // changed facilityID in all classes to type UUID.
+        User newUser = new FacilityUser(username, password, facilityID, type);
```

Comment:
The addNewUser method should be returning some sort of NewUserResponseModel object which will have information about success/fail, etc. That way we can use it in the presenter to display the result of the creation. Same in the controller. We need to add input/output interfaces and data models for this.